lib/async_hooks.js

```diff
@@ -365,21 +365,14 @@ function emitBeforeN(asyncId) {
 
 // Usage: emitBeforeS(asyncId[, triggerAsyncId]). If triggerAsyncId is omitted
 // then asyncId will be used instead.
-function emitBeforeS(asyncId, triggerAsyncId = asyncId) {
+function emitBeforeS(asyncId, triggerAsyncId) {
   // CHECK(Number.isSafeInteger(asyncId) && asyncId > 0)
   // CHECK(Number.isSafeInteger(triggerAsyncId) && triggerAsyncId > 0)
 
-  // Validate the ids.
-  if (asyncId < 0 || triggerAsyncId < 0) {
-    fatalError('before(): asyncId or triggerAsyncId is less than zero ' +
-               `(asyncId: ${asyncId}, triggerAsyncId: ${triggerAsyncId})`);
-  }
```

Comment:
`emitBeforeS` is a (undocumented) user API and is exported via `require()`. The general rule has been that it shouldn't be possible for the application to abort via any method imported via `require()` (this does not hold true for `process.binding()`). But removing this allows `require('async_hooks').emitBefore(-1, -1)` to cause a `CHECK()` failure.